ttnn/cpp/ttnn/operations/moreh/moreh_linear_backward/device/moreh_linear_backward_device_operation.cpp

```diff
@@ -0,0 +1,71 @@
+// SPDX-FileCopyrightText: © 2023 Tenstorrent Inc.
+//
+// SPDX-License-Identifier: Apache-2.0
+
+#include "moreh_linear_backward_device_operation.hpp"
+
+#include <cstdint>
+
+#include "tt_dnn/op_library/moreh_helper_functions.hpp"
+#include "ttnn/operations/core/compute_kernel/compute_kernel_config.hpp"
+#include "ttnn/tensor/tensor.hpp"
+
+namespace ttnn::operations::moreh::moreh_linear_backward {
+
+MorehBiasAddBackwardOperation::program_factory_t MorehBiasAddBackwardOperation::select_program_factory(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+    const auto& bias_grad = tensor_args.bias_grad.value();
+    if (tt::operations::primary::is_scalar(bias_grad))
+        return SingleCoreProgramFactory();
+    return MultiCoreProgramFactory();
+}
+
+void MorehBiasAddBackwardOperation::validate_on_program_cache_miss(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+};
+
+void MorehBiasAddBackwardOperation::validate_on_program_cache_hit(
+    const operation_attributes_t& operation_attributes, const tensor_args_t& tensor_args) {
+};
```

Comment:
I think validation is also necessary for primitives. 
I know that the composite version of `linear backward` has validation, but the primitive `bias_add_backward` can also be called directly from other code, so validation should be implemented for it as well.

Reply:
resolved in [ae00a8b](https://github.com/tenstorrent/tt-metal/pull/12812/commits/ae00a8bdcfd5f6462e3a0867c135a0661f64ca85)